src/test/java/gift/controller/ProductControllerTest.java

```diff
@@ -0,0 +1,88 @@
+package gift.controller;
+
+import gift.model.Product;
+import gift.model.ProductDao;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+class ProductControllerTest {
+
+    @Mock
+    private ProductDao productDao;
+
+    @InjectMocks
+    private ProductController productController;
+
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        mockMvc = MockMvcBuilders.standaloneSetup(productController).build();
+    }
+
+    @Test
+    void testGetProducts() throws Exception {
+        // given: 테스트에 사용될 mock 데이터를 설정합니다.
+        var products = Arrays.asList(
```

Comment:
List.of를 사용하면 불변 + 최적화가 된 형태로 리스트 생성이 가능합니다ㅎㅎ
```suggestion
        var products = List.of(
```

Reply:
넵 감사합니다!